Kaguya/Kaguya/Discord/Commands/Reference/Osu.cs

```diff
@@ -0,0 +1,121 @@
+﻿using Discord.Commands;
+using Kaguya.Discord.Attributes;
+using Microsoft.Extensions.Logging;
+using System.Threading.Tasks;
+using Discord;
+using Kaguya.Database.Model;
+using Kaguya.Database.Repositories;
+using Kaguya.Discord.DiscordExtensions;
+using Kaguya.External.Osu;
+using OsuSharp;
+
+namespace Kaguya.Discord.Commands.Reference
+{
+    [Module(CommandModule.Reference)]
+    [Group("osu")]
+    [Alias("o")]
+    public class Osu : KaguyaBase<Osu>
+    {
+        private readonly ILogger<Osu> _logger;
+        private readonly OsuClient _osuClient;
+        private readonly KaguyaUserRepository _kaguyaUserRepository;
+        private readonly KaguyaServerRepository _kaguyaServerRepository;
+        private const string GAME_MODE_STRING = "**Valid Game Modes:** `std`, `mania`, `taiko`, `ctb`";
+
+        public Osu(ILogger<Osu> logger, OsuClient osuClient, KaguyaUserRepository kaguyaUserRepository, 
+            KaguyaServerRepository kaguyaServerRepository) : base(logger)
+        {
+            _logger = logger;
+            _osuClient = osuClient;
+            _kaguyaUserRepository = kaguyaUserRepository;
+            _kaguyaServerRepository = kaguyaServerRepository;
+        }
+
+        [Priority(1)]
+        [Command("-set")]
+        [Summary("Give me an osu! username to remember you by! After setting a username, you will no longer have " +
+                 "to specify a username when using osu commands. You must also configure your game mode.\n\n" + GAME_MODE_STRING)]
+        [Remarks("<game mode> <username or id>")]
+        [Example("std name with spaces")]
+        [Example("mania 99999999")]
+        [Example("ctb really good ctb player")]
+        public async Task OsuSetCommand(string gameModeString, [Remainder] string username)
```

Comment:
should gameModeString have input validation? there is no input validation in this method